Jenkinsfile

```diff
@@ -178,8 +178,8 @@ pipeline {
                     steps {
                         deleteDir()
                         unstash 'MathSetup'
-                        sh "echo CC=${env.CXX} -Werror > make/local"
-                        sh "echo CXXFLAGS+=-DSTAN_THREADS >> make/local"
+                        sh "echo CXX=${env.CXX} -Werror > make/local"
+                        sh "echo CPPFLAGS=-DSTAN_THREADS >> make/local"
```

Comment:
I don't understand.

The use of `CXXFLAGS` here is just abused. `CXX` refers to the compiler, `CPP` refers to the C preprocessor. `-DSTAN_THREADS` should really be set in the preprocessor options, not compiler options. We can abuse it and be ok, but we should just do the right thing if we can. It'll make it easier for a future developer to google or search on StackOverflow because we're using make correctly. It'll turn up more relevant answers.

Reply:
Oh. I did not know that. Still, should we do CPPFLAGS+=... as a default? Maybe we want to pre-define CPPFLAGS or the system as setup some. Maybe not relevant for Jenksins though.

Reply:
Here, we want to override the CPPFLAGS. It might make sense later on to add a variable for threading, but we should make that a separate pull request.